### ISSUE TITLE
what is 0.8 in def train_d?

### ISSUE THREAD

**Issue by sarrbranka** [odegeasslbc/FastGAN-pytorch#13]
what is 0.8 in def train_d?  How did you choose the number?

https://github.com/odegeasslbc/FastGAN-pytorch/blob/902a47397ae6a1dc23142aeda62ebaa00d5e87d2/train.py#L39

**Comment by odegeasslbc** [comment]
This is a simply smoothed training loss. A traditional hinge loss pushes the output of D from real images to be 1 and from fake images to be 0. But here we relax the hard boundary of 1 and make it random (between 0.8 and 1). It prevents D from getting overpowered, also adds some perturbation to the training. In practice, I found the impact is not big, so you may remove it. I do not specifically search for the best number, 0.8 is me randomly picked
